homework/advanced-calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,194 @@
+#include "advancedCalculator.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <cctype>
+#include <cmath>
+#include <numeric>
+#include <vector>
+
+std::map<const char, std::function<double(const double, const double)>> operations;
+std::string allOperations { "+-*/%!^$ " };
+double firstValue { };
+char operation { };
+double secondValue { };
+double secondMinus { 1 };
+bool oneValue { false };
+bool isFactorial { false };
```

Comment:
ugly.

Maybe you should wrap it to some class? Don't use global vlaues like in C :)